tsgo reports additional type error when a property is followed by an auto-accessor of the same name

## Steps to reproduce

`main.ts`:
```ts
class C {
  y: number = 2;
  accessor y: number = 3;
}
```

```sh
tsgo main.ts
```

## Behavior with `typescript@6.0`

```text
main.ts(2,3): error TS2300: Duplicate identifier 'y'.
main.ts(3,12): error TS2300: Duplicate identifier 'y'.
```

## Behavior with `tsgo`

```text
main.ts(2,3): error TS2300: Duplicate identifier 'y'.
main.ts(3,12): error TS2300: Duplicate identifier 'y'.
main.ts(3,12): error TS2717: Subsequent property declarations must have the same type.  Property 'y' must be of type 'any', but here has type 'number'.
```

--- a/internal/checker/checker.go
+++ b/internal/checker/checker.go
@@ -18374,11 +18374,7 @@ func (c *Checker) getTypeOfAccessors(symbol *ast.Symbol) *Type {
 		}
 		getter := ast.GetDeclarationOfKind(symbol, ast.KindGetAccessor)
 		setter := ast.GetDeclarationOfKind(symbol, ast.KindSetAccessor)
-		property := ast.GetDeclarationOfKind(symbol, ast.KindPropertyDeclaration)
-		var accessor *ast.Node
-		if property != nil && ast.IsAutoAccessorPropertyDeclaration(property) {
-			accessor = property
-		}
+		accessor := core.Find(symbol.Declarations, ast.IsAutoAccessorPropertyDeclaration)
 		// We try to resolve a getter type annotation, a setter type annotation, or a getter function
 		// body return type inference, in that order.
 		t := c.getAnnotatedAccessorType(getter)

--- a/testdata/baselines/reference/compiler/duplicatePropertyAndAccessor.errors.txt
+++ b/testdata/baselines/reference/compiler/duplicatePropertyAndAccessor.errors.txt
@@ -1,0 +1,16 @@
+duplicatePropertyAndAccessor.ts(4,3): error TS2300: Duplicate identifier 'y'.
+duplicatePropertyAndAccessor.ts(5,12): error TS2300: Duplicate identifier 'y'.
+
+
+==== duplicatePropertyAndAccessor.ts (2 errors) ====
+    // https://github.com/microsoft/typescript-go/issues/4130
+    
+    class C {
+      y: number = 2;
+      ~
+!!! error TS2300: Duplicate identifier 'y'.
+      accessor y: number = 3;
+               ~
+!!! error TS2300: Duplicate identifier 'y'.
+    }
+    

--- a/testdata/baselines/reference/compiler/duplicatePropertyAndAccessor.js
+++ b/testdata/baselines/reference/compiler/duplicatePropertyAndAccessor.js
@@ -1,0 +1,25 @@
+//// [tests/cases/compiler/duplicatePropertyAndAccessor.ts] ////
+
+//// [duplicatePropertyAndAccessor.ts]
+// https://github.com/microsoft/typescript-go/issues/4130
+
+class C {
+  y: number = 2;
+  accessor y: number = 3;
+}
+
+
+//// [duplicatePropertyAndAccessor.js]
+"use strict";
+// https://github.com/microsoft/typescript-go/issues/4130
+class C {
+    y = 2;
+    accessor y = 3;
+}
+
+
+//// [duplicatePropertyAndAccessor.d.ts]
+declare class C {
+    y: number;
+    accessor y: number;
+}

--- a/testdata/baselines/reference/compiler/duplicatePropertyAndAccessor.symbols
+++ b/testdata/baselines/reference/compiler/duplicatePropertyAndAccessor.symbols
@@ -1,0 +1,15 @@
+//// [tests/cases/compiler/duplicatePropertyAndAccessor.ts] ////
+
+=== duplicatePropertyAndAccessor.ts ===
+// https://github.com/microsoft/typescript-go/issues/4130
+
+class C {
+>C : Symbol(C, Decl(duplicatePropertyAndAccessor.ts, 0, 0))
+
+  y: number = 2;
+>y : Symbol(C.y, Decl(duplicatePropertyAndAccessor.ts, 2, 9), Decl(duplicatePropertyAndAccessor.ts, 3, 16))
+
+  accessor y: number = 3;
+>y : Symbol(C.y, Decl(duplicatePropertyAndAccessor.ts, 2, 9), Decl(duplicatePropertyAndAccessor.ts, 3, 16))
+}
+

--- a/testdata/baselines/reference/compiler/duplicatePropertyAndAccessor.types
+++ b/testdata/baselines/reference/compiler/duplicatePropertyAndAccessor.types
@@ -1,0 +1,17 @@
+//// [tests/cases/compiler/duplicatePropertyAndAccessor.ts] ////
+
+=== duplicatePropertyAndAccessor.ts ===
+// https://github.com/microsoft/typescript-go/issues/4130
+
+class C {
+>C : C
+
+  y: number = 2;
+>y : number
+>2 : 2
+
+  accessor y: number = 3;
+>y : number
+>3 : 3
+}
+

--- a/testdata/tests/cases/compiler/duplicatePropertyAndAccessor.ts
+++ b/testdata/tests/cases/compiler/duplicatePropertyAndAccessor.ts
@@ -1,0 +1,9 @@
+// @declaration: true
+// @target: esnext
+
+// https://github.com/microsoft/typescript-go/issues/4130
+
+class C {
+  y: number = 2;
+  accessor y: number = 3;
+}

